src/precice/impl/DataContext.hpp

```diff
@@ -6,78 +6,170 @@
+   * 
+   * @return int ID of _toData.
+   */
   int getToDataID() const;
 
+  /**
+   * @brief Get the dimensions of _providedData.
+   * 
+   * @return int Dimensions of _providedData.
+   */
+  int getDataDimensions() const;
+
+  /**
+   * @brief Get the name of _mesh.
+   * 
+   * @return std::string Name of _mesh.
+   */
   std::string getMeshName() const;
 
+  /**
+   * @brief Get the ID of _mesh.
+   * 
+   * @return int ID of _mesh.
+   */
   int getMeshID() const;
 
   /**
-   * @brief links a read mapping and the mesh context the read mapping requires to this data context
-   *
-   * @param[in] mappingContext provides context of read mapping
-   * @param[in] fromMeshContext provides context of mesh this read mapping is mapping to (_toData)
+   * @brief Informs the user whether this DataContext has a _mappingContext.
+   * 
+   * @return True, if this DataContext is associated with a mapping. False, if not. 
    */
-  void configureForReadMapping(MappingContext mappingContext, MeshContext fromMeshContext);
+  bool hasMapping() const;
 
   /**
-   * @brief links a write mapping and the mesh context the write mapping requires to this data context
-   *
-   * @param[in] mappingContext provides context of write mapping
-   * @param[in] fromMeshContext provides context of mesh this write mapping is mapping from (_fromData)
+   * @brief Get the _mappingContext associated with this DataContext.
+   * 
+   * @return const MappingContext The _mappingContext of this DataContext.
    */
-  void configureForWriteMapping(MappingContext mappingContext, MeshContext toMeshContext);
-
-  bool hasMapping() const;
+  const MappingContext mappingContext() const;
 
+  /**
+   * @brief Informs the user whether this DataContext has a read mapping.
+   * 
+   * @return True, if DataContext has a read mapping.
+   */
   bool hasReadMapping() const;
 
+  /**
+   * @brief Informs the user whether this DataContext has a write mapping.
+   * 
+   * @return True, if DataContext has a write mapping.
+   */
   bool hasWriteMapping() const;
 
-  const MappingContext mappingContext() const;
+  /**
+   * @brief Links a MappingContext and the MeshContext required by the mapping to this DataContext.
+   * 
+   * A mapping maps the given data from or to _providedData (depending on whether it is a read or write mapping).
+   *
+   * @param[in] mappingContext Context of read mapping
+   * @param[in] meshContext Context of mesh this mapping is mapping from or to
+   */
+  virtual void configureMapping(MappingContext mappingContext, MeshContext meshContext) = 0;
 
-private:
-  mesh::PtrMesh _mesh;
+protected:
+  /**
+   * @brief Construct a new DataContext without a mapping. Protected, because only ReadDataContext and WriteDataContext should use this constructor.
+   * 
+   * @param data Data associated with this DataContext.
+   * @param mesh Mesh associated with this DataContext.
+   */
+  DataContext(mesh::PtrData data, mesh::PtrMesh mesh);
```

Comment:
Not sure why this is protected. This class is a abstract class and cannot be instantiated. ( `=0`)

Reply:
Does this maybe help?

```
ReadDataContext::ReadDataContext(
    mesh::PtrData data,
    mesh::PtrMesh mesh)
    : DataContext(data, mesh)
{
}
```

`ReadDataContext` and `WriteDataContext` use the constructor of `DataContext`:

```
DataContext::DataContext(mesh::PtrData data, mesh::PtrMesh mesh)
{
  PRECICE_TRACE();
  PRECICE_ASSERT(data);
  _providedData = data;
  PRECICE_ASSERT(mesh);
  _mesh = mesh;
}
```

I could shift these few lines of code directly into the constructors of the children classes. But then I also have to make `_mesh` protected. I think it is nice that only `DataContext` knows the `_mesh` and for the children `_mesh` is inaccessible.